### PR TITLE
Use github.com/santhosh-tekuri/jsonschema/v5

### PR DESCRIPTION
**Pull request by zregvart**
The github.com/qri-io/jsonschema seems to suffer from thread safety issues, see[1]. This replaces that implementation with github.com/santhosh-tekuri/jsonschema/v5.

Note that the `uniqueKeys` functionality that has been disabled in #1534 has not been ported and since it is now dead code has been removed.

[1] https://github.com/enterprise-contract/ec-cli/actions/runs/8777708975/job/24083046629?pr=1544#step:6:5121

### CODE REVIEW

File: internal/evaluation_target/application_snapshot_image/application_snapshot_image.go [hunk 1]
```diff
@@ -252,41 +252,41 @@ func (a ApplicationSnapshotImage) ValidateAttestationSyntax(ctx context.Context)
...
+	var validationErr *jsonschema.ValidationError
 	for _, sp := range a.attestations {
 		pt := sp.PredicateType()
 		if schema, ok := attestationSchemas[pt]; ok {
 			// Found a validator for this predicate type so let's use it
 			log.Debugf("Attempting to validate an attestation with predicateType %s", pt)
-			if errs, err := schema.ValidateBytes(ctx, sp.Statement()); err != nil {
-				// Error while trying to validate
+
+			var st any
```
Comment (robnester-rh) [inline internal/evaluation_target/application_snapshot_image/application_snapshot_image.go:262]: Perhaps a comment to explain what `st` is, for code maintainability?

File: internal/evaluation_target/application_snapshot_image/application_snapshot_image.go [hunk 2]
```diff
@@ -252,41 +252,41 @@ func (a ApplicationSnapshotImage) ValidateAttestationSyntax(ctx context.Context)
...
-			} else {
-				if len(errs) == 0 {
-					log.Debugf("Statement schema was validated successfully against the %s schema", pt)
-				} else {
-					log.Debugf("Validated the statement against %s schema and found the following errors: %v", pt, errs)
-					allErrors[pt] = errs
+			}
+
+			if err := schema.Validate(st); err != nil {
+				if validationErr, ok = err.(*jsonschema.ValidationError); !ok {
```
Comment (lcarva) [inline internal/evaluation_target/application_snapshot_image/application_snapshot_image.go:268]: If there are two attestations, wouldn't the schema validation errors on the first one be thrown out if the second one validates successfully?

File: internal/evaluation_target/application_snapshot_image/application_snapshot_image.go (hunk 1 above)
Comment (zregvart) [reply internal/evaluation_target/application_snapshot_image/application_snapshot_image.go]: Renamed to `statement`

File: internal/evaluation_target/application_snapshot_image/application_snapshot_image.go (hunk 2 above)
Comment (zregvart) [reply internal/evaluation_target/application_snapshot_image/application_snapshot_image.go]: Now collects all `jsonschema.ValidationError` errors, similar to how it was done prior

**Comment by zregvart** [comment]
I think review comments were addressed, will merge now.

**Pull request merged**